src/dodal/devices/focusing_mirror.py

```diff
@@ -107,30 +114,49 @@ def voltage_channels(self) -> list[MirrorVoltageDevice]:
         ]
 
 
-class FocusingMirror(Device):
+class FocusingMirror(StandardReadable):
     """Focusing Mirror"""
 
-    def __init__(self, bragg_to_lat_lut_path, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(
+        self, name, prefix, bragg_to_lat_lut_path=None, x_y_sufffixes=("X", "Y")
+    ):
         self.bragg_to_lat_lookup_table_path = bragg_to_lat_lut_path
+        self.yaw_mrad = Motor(prefix + "YAW")
+        self.pitch_mrad = Motor(prefix + "PITCH")
+        self.roll_mrad = Motor(prefix + "ROLL")
+        self.x_mm = Motor(prefix + x_y_sufffixes[0])
+        self.y_mm = Motor(prefix + x_y_sufffixes[1])
+        self.jack1_mm = Motor(prefix + "Y1")
+        self.jack2_mm = Motor(prefix + "Y2")
+        self.jack3_mm = Motor(prefix + "Y3")
+        self.translation1_mm = Motor(prefix + "X1")
+        self.translation2_mm = Motor(prefix + "X2")
+
+        # See https://manual.nexusformat.org/classes/base_classes/NXmirror.html
+        self.type, _ = soft_signal_r_and_backend(str, "single")
+        # The device is in the beamline co-ordinate system so pitch is the incident angle
+        # regardless of orientation of the mirror
+        self.incident_angle = Motor(prefix + "PITCH")
```

Comment:
Maybe you've thought about this already but is there a reason for this over
```suggestion
        self.incident_angle = self.pitch_mrad
```

?

Reply:
I thought about that but if you have one signal like that I'm not sure how the name gets set on it? Probably would get the last one set so might be alright but then would you get `incident_angle` when you read `pitch`?

To be honest, I'm not sure of the best way to structure this, I assume you want the name to be `incident_angle` for the nexus writer?

Reply:
Yep fair enough sounds like I just missed some complexity (naming the signals) - feel free to leave as it was.